app/models/course_user.rb

```diff
@@ -33,7 +33,7 @@ def experience_points
       sum(:points_awarded)
     end
   end
-  has_one :invitation, class_name: Course::UserInvitation.name, dependent: :destroy
+  has_one :invitation, class_name: Course::UserInvitation.name, dependent: :destroy, validate: true
```

Comment:
I think we should merge this first, since we don't have delete. When we need delete hopefully it's fixed in Rails.